.gitignore

```diff
@@ -1,5 +1,20 @@
+.idea
+vendor/
+
+# composer
 composer.lock
 composer.phar
-vendor/
+
+# osx
 .DS_Store
-.idea
+.DS_Store?
+._*
+.Spotlight-V100
+.Trashes
+
+# npm
+.npm/node_modules
+node_modules
+
+# generated files
+templates/theme/css
```

Comment:
In order:
* osx files generated when using spotlight/doing changes by using finder gui and, i believe, when moving things to trash in finder gui.
* modules downloaded by npm when doing npm install.
* generated css files which should not be kept in the repo.

addressing vendor and idea in this comment since they've been split up -- the two entries were not removed, but moved to the top of the .gitignore file.

does that not make sense? for me it does, but please correct me if I'm wrong